### PR TITLE
python packages updates

### PR DESCRIPTION
**Pull request by happysalada**
## Description of changes

I'm still having a weird error on startup on darwin
```
/nix/store/1zaap1xxxvw2ypsgh1mfxb3wzdd49873-python3-3.11.8/bin/python3.11: No module named ipykernel_launcher
```

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by happysalada** [comment]
@natsukium sorry to tag you, but if you have any idea about the missing ipython_kernel module failure, I'm happy to test anything.

**Comment by natsukium** [comment]
I'll look at it tonight (a few hours later).

**Comment by natsukium** [comment]
It seems that unwrapped Python is called somewhere. It may work if you explicitly set PYTHONPATH.

```
  makeWrapperArgs = [
    "--prefix PYTHONPATH : ${python3.pkgs.makePythonPath propagatedBuildInputs}"
  ];
```

**Comment by happysalada** [comment]
@natsukium you're amazing! Thank you!

**Comment by drupol** [comment]
Had to do the same in https://github.com/NixOS/nixpkgs/pull/299738 too !

**Comment by happysalada** [comment]
I'm going to drop the open-interpreter update, and will just wait for ofborg to finish eval and merge just the python packages update. I'm getting errors when trying to make queries with open interpreter.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 298486` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>chatgpt-retrieval-plugin</li>
    <li>chatgpt-retrieval-plugin.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
  </ul>
</details>
<details>
  <summary>189 packages built:</summary>
  <ul>
    <li>chatblade</li>
    <li>chatblade.dist</li>
    <li>checkov</li>
    <li>checkov.dist</li>
    <li>home-assistant-component-tests.openai_conversation</li>
    <li>imaginer</li>
    <li>llm (python311Packages.llm)</li>
    <li>llm.dist (python311Packages.llm.dist)</li>
    <li>open-interpreter (python311Packages.open-interpreter)</li>
    <li>open-interpreter.dist (python311Packages.open-interpreter.dist)</li>
    <li>openai (python311Packages.openai)</li>
    <li>openai.dist (python311Packages.openai.dist)</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python311Packages.aifs</li>
    <li>python311Packages.aifs.dist</li>
    <li>python311Packages.clarifai</li>
    <li>python311Packages.clarifai.dist</li>
    <li>python311Packages.guidance</li>
    <li>python311Packages.guidance.dist</li>
    <li>python311Packages.instructor</li>
    <li>python311Packages.instructor.dist</li>
    <li>python311Packages.jsonpath-python</li>
    <li>python311Packages.jsonpath-python.dist</li>
    <li>python311Packages.litellm</li>
    <li>python311Packages.litellm.dist</li>
    <li>python311Packages.llama-index</li>
    <li>python311Packages.llama-index-agent-openai</li>
    <li>python311Packages.llama-index-agent-openai.dist</li>
    <li>python311Packages.llama-index-cli</li>
    <li>python311Packages.llama-index-cli.dist</li>
    <li>python311Packages.llama-index-core</li>
    <li>python311Packages.llama-index-core.dist</li>
    <li>python311Packages.llama-index-embeddings-gemini</li>
    <li>python311Packages.llama-index-embeddings-gemini.dist</li>
    <li>python311Packages.llama-index-embeddings-google</li>
    <li>python311Packages.llama-index-embeddings-google.dist</li>
    <li>python311Packages.llama-index-embeddings-huggingface</li>
    <li>python311Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python311Packages.llama-index-embeddings-ollama</li>
    <li>python311Packages.llama-index-embeddings-ollama.dist</li>
    <li>python311Packages.llama-index-embeddings-openai</li>
    <li>python311Packages.llama-index-embeddings-openai.dist</li>
    <li>python311Packages.llama-index-graph-stores-nebula</li>
    <li>python311Packages.llama-index-graph-stores-nebula.dist</li>
    <li>python311Packages.llama-index-graph-stores-neo4j</li>
    <li>python311Packages.llama-index-graph-stores-neo4j.dist</li>
    <li>python311Packages.llama-index-graph-stores-neptune</li>
    <li>python311Packages.llama-index-graph-stores-neptune.dist</li>
    <li>python311Packages.llama-index-indices-managed-llama-cloud</li>
    <li>python311Packages.llama-index-indices-managed-llama-cloud.dist</li>
    <li>python311Packages.llama-index-legacy</li>
    <li>python311Packages.llama-index-legacy.dist</li>
    <li>python311Packages.llama-index-llms-ollama</li>
    <li>python311Packages.llama-index-llms-ollama.dist</li>
    <li>python311Packages.llama-index-llms-openai</li>
    <li>python311Packages.llama-index-llms-openai-like</li>
    <li>python311Packages.llama-index-llms-openai-like.dist</li>
    <li>python311Packages.llama-index-llms-openai.dist</li>
    <li>python311Packages.llama-index-multi-modal-llms-openai</li>
    <li>python311Packages.llama-index-multi-modal-llms-openai.dist</li>
    <li>python311Packages.llama-index-program-openai</li>
    <li>python311Packages.llama-index-program-openai.dist</li>
    <li>python311Packages.llama-index-question-gen-openai</li>
    <li>python311Packages.llama-index-question-gen-openai.dist</li>
    <li>python311Packages.llama-index-readers-database</li>
    <li>python311Packages.llama-index-readers-database.dist</li>
    <li>python311Packages.llama-index-readers-file</li>
    <li>python311Packages.llama-index-readers-file.dist</li>
    <li>python311Packages.llama-index-readers-json</li>
    <li>python311Packages.llama-index-readers-json.dist</li>
    <li>python311Packages.llama-index-readers-llama-parse</li>
    <li>python311Packages.llama-index-readers-llama-parse.dist</li>
    <li>python311Packages.llama-index-readers-s3</li>
    <li>python311Packages.llama-index-readers-s3.dist</li>
    <li>python311Packages.llama-index-readers-twitter</li>
    <li>python311Packages.llama-index-readers-twitter.dist</li>
    <li>python311Packages.llama-index-readers-txtai</li>
    <li>python311Packages.llama-index-readers-txtai.dist</li>
    <li>python311Packages.llama-index-readers-weather</li>
    <li>python311Packages.llama-index-readers-weather.dist</li>
    <li>python311Packages.llama-index-vector-stores-chroma</li>
    <li>python311Packages.llama-index-vector-stores-chroma.dist</li>
    <li>python311Packages.llama-index-vector-stores-google</li>
    <li>python311Packages.llama-index-vector-stores-google.dist</li>
    <li>python311Packages.llama-index-vector-stores-postgres</li>
    <li>python311Packages.llama-index-vector-stores-postgres.dist</li>
    <li>python311Packages.llama-index-vector-stores-qdrant</li>
    <li>python311Packages.llama-index-vector-stores-qdrant.dist</li>
    <li>python311Packages.llama-index.dist</li>
    <li>python311Packages.llama-parse</li>
    <li>python311Packages.llama-parse.dist</li>
    <li>python311Packages.python-iso639</li>
    <li>python311Packages.python-iso639.dist</li>
    <li>python311Packages.unstructured</li>
    <li>python311Packages.unstructured-client</li>
    <li>python311Packages.unstructured-client.dist</li>
    <li>python311Packages.unstructured.dist</li>
    <li>python312Packages.aifs</li>
    <li>python312Packages.aifs.dist</li>
    <li>python312Packages.clarifai</li>
    <li>python312Packages.clarifai.dist</li>
    <li>python312Packages.guidance</li>
    <li>python312Packages.guidance.dist</li>
    <li>python312Packages.jsonpath-python</li>
    <li>python312Packages.jsonpath-python.dist</li>
    <li>python312Packages.litellm</li>
    <li>python312Packages.litellm.dist</li>
    <li>python312Packages.llama-index</li>
    <li>python312Packages.llama-index-agent-openai</li>
    <li>python312Packages.llama-index-agent-openai.dist</li>
    <li>python312Packages.llama-index-cli</li>
    <li>python312Packages.llama-index-cli.dist</li>
    <li>python312Packages.llama-index-core</li>
    <li>python312Packages.llama-index-core.dist</li>
    <li>python312Packages.llama-index-embeddings-gemini</li>
    <li>python312Packages.llama-index-embeddings-gemini.dist</li>
    <li>python312Packages.llama-index-embeddings-google</li>
    <li>python312Packages.llama-index-embeddings-google.dist</li>
    <li>python312Packages.llama-index-embeddings-huggingface</li>
    <li>python312Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python312Packages.llama-index-embeddings-ollama</li>
    <li>python312Packages.llama-index-embeddings-ollama.dist</li>
    <li>python312Packages.llama-index-embeddings-openai</li>
    <li>python312Packages.llama-index-embeddings-openai.dist</li>
    <li>python312Packages.llama-index-graph-stores-nebula</li>
    <li>python312Packages.llama-index-graph-stores-nebula.dist</li>
    <li>python312Packages.llama-index-graph-stores-neo4j</li>
    <li>python312Packages.llama-index-graph-stores-neo4j.dist</li>
    <li>python312Packages.llama-index-graph-stores-neptune</li>
    <li>python312Packages.llama-index-graph-stores-neptune.dist</li>
    <li>python312Packages.llama-index-indices-managed-llama-cloud</li>
    <li>python312Packages.llama-index-indices-managed-llama-cloud.dist</li>
    <li>python312Packages.llama-index-legacy</li>
    <li>python312Packages.llama-index-legacy.dist</li>
    <li>python312Packages.llama-index-llms-ollama</li>
    <li>python312Packages.llama-index-llms-ollama.dist</li>
    <li>python312Packages.llama-index-llms-openai</li>
    <li>python312Packages.llama-index-llms-openai-like</li>
    <li>python312Packages.llama-index-llms-openai-like.dist</li>
    <li>python312Packages.llama-index-llms-openai.dist</li>
    <li>python312Packages.llama-index-multi-modal-llms-openai</li>
    <li>python312Packages.llama-index-multi-modal-llms-openai.dist</li>
    <li>python312Packages.llama-index-program-openai</li>
    <li>python312Packages.llama-index-program-openai.dist</li>
    <li>python312Packages.llama-index-question-gen-openai</li>
    <li>python312Packages.llama-index-question-gen-openai.dist</li>
    <li>python312Packages.llama-index-readers-database</li>
    <li>python312Packages.llama-index-readers-database.dist</li>
    <li>python312Packages.llama-index-readers-file</li>
    <li>python312Packages.llama-index-readers-file.dist</li>
    <li>python312Packages.llama-index-readers-json</li>
    <li>python312Packages.llama-index-readers-json.dist</li>
    <li>python312Packages.llama-index-readers-llama-parse</li>
    <li>python312Packages.llama-index-readers-llama-parse.dist</li>
    <li>python312Packages.llama-index-readers-s3</li>
    <li>python312Packages.llama-index-readers-s3.dist</li>
    <li>python312Packages.llama-index-readers-twitter</li>
    <li>python312Packages.llama-index-readers-twitter.dist</li>
    <li>python312Packages.llama-index-readers-txtai</li>
    <li>python312Packages.llama-index-readers-txtai.dist</li>
    <li>python312Packages.llama-index-readers-weather</li>
    <li>python312Packages.llama-index-readers-weather.dist</li>
    <li>python312Packages.llama-index-vector-stores-chroma</li>
    <li>python312Packages.llama-index-vector-stores-chroma.dist</li>
    <li>python312Packages.llama-index-vector-stores-google</li>
    <li>python312Packages.llama-index-vector-stores-google.dist</li>
    <li>python312Packages.llama-index-vector-stores-postgres</li>
    <li>python312Packages.llama-index-vector-stores-postgres.dist</li>
    <li>python312Packages.llama-index-vector-stores-qdrant</li>
    <li>python312Packages.llama-index-vector-stores-qdrant.dist</li>
    <li>python312Packages.llama-index.dist</li>
    <li>python312Packages.llama-parse</li>
    <li>python312Packages.llama-parse.dist</li>
    <li>python312Packages.open-interpreter</li>
    <li>python312Packages.open-interpreter.dist</li>
    <li>python312Packages.openai</li>
    <li>python312Packages.openai.dist</li>
    <li>python312Packages.python-iso639</li>
    <li>python312Packages.python-iso639.dist</li>
    <li>python312Packages.unstructured</li>
    <li>python312Packages.unstructured-client</li>
    <li>python312Packages.unstructured-client.dist</li>
    <li>python312Packages.unstructured.dist</li>
    <li>shell-genie</li>
    <li>shell-genie.dist</li>
    <li>shell_gpt</li>
    <li>shell_gpt.dist</li>
    <li>unstructured-api</li>
  </ul>
</details>

File: pkgs/development/python-modules/jsonpath-python/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  src = fetchFromGitHub {
+    owner = "sean2077";
+    repo = "jsonpath-python";
+    rev = "aead1e4db9890f12fc029ca57b9732aa16c62915";
+    hash = "sha256-q1wjtW1USn88ISYDTnT6DA57WmZeUP25U2zrR8B/Rfk=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
```
Comment (dotlambda) [inline pkgs/development/python-modules/jsonpath-python/default.nix:22]:
```suggestion
```
not needed

File: pkgs/development/python-modules/jsonpath-python/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+buildPythonPackage rec {
+  pname = "jsonpath-python";
+  version = "1.0.6";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "sean2077";
+    repo = "jsonpath-python";
+    rev = "aead1e4db9890f12fc029ca57b9732aa16c62915";
```
Comment (dotlambda) [inline pkgs/development/python-modules/jsonpath-python/default.nix:16]: This warrants a comment as well.

File: pkgs/development/python-modules/python-iso639/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    hash = "sha256-ANZV/Q+2V5m5/7/1qkw2gRCaimRMlvXhOrewe5wmP1s=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
+  ];
+
+  passthru.optional-dependencies = {
+    dev = [
```
Comment (dotlambda) [inline pkgs/development/python-modules/python-iso639/default.nix:31]: No need for this cause it's only used for testing and development.

File: pkgs/development/python-modules/aifs/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "OpenInterpreter";
+    repo = "aifs";
+    rev = "912547bfa576f4f8b9f8908b15179173376d630e";
+    hash = "sha256-ki84Ls14jcL6OrvFZmmZfQKReVg/yKqDLXXumVsLrsY=";
+  };
+
+  nativeBuildInputs = [
```
Comment (dotlambda) [inline pkgs/development/python-modules/aifs/default.nix:22]: We should use
```suggestion
  build-system = [
```
...

File: pkgs/development/python-modules/aifs/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
```
Comment (dotlambda) [inline pkgs/development/python-modules/aifs/default.nix:26]: ... and
```suggestion
  dependencies = [
```
nowadays.

**Review by dotlambda — COMMENTED** [review]
Lots of packages are missing `nativeCheckInputs` (or `doCheck = false` in case there are no tests).

**Comment by happysalada** [comment]
Alright I messed up this branch and created a new one with the fixes suggested
https://github.com/NixOS/nixpkgs/pull/300481

**Pull request closed without merging**